### PR TITLE
feat(risedev): generate docker-compose.yaml

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

```
./risedev compose <profile>
```

will generate a docker-compose.yaml

Currently, only compute, meta and frontend nodes are supported. We will add more in the future. For benchmarking, we should generate `network: host` so as to bypass docker network stack. There will be an option controlling generating behavior in the future.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/2285

### CODE REVIEW

**Comment by skyzh** [comment]
An example of generated yaml:

```yaml
---
version: "3"
services:
  meta-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - "--host"
      - "0.0.0.0:5690"
      - "--dashboard-host"
      - "0.0.0.0:5691"
      - "--prometheus-host"
      - "0.0.0.0:1250"
      - "--backend"
      - mem
      - "--disable-recovery"
    expose:
      - "5690"
      - "1250"
      - "5691"
    ports: []
    depends_on: []
  compute-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - "--host"
      - "0.0.0.0:5688"
      - "--prometheus-listener-addr"
      - "0.0.0.0:1222"
      - "--client-address"
      - "compute-node-0:5688"
      - "--metrics-level"
      - "1"
      - "--state-store"
      - hummock+memory
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "5688"
      - "1222"
    ports: []
    depends_on:
      - meta-node-0
  frontend-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - "--host"
      - "0.0.0.0:4566"
      - "--meta-addr"
      - "http://meta-node-0:5690"
    expose:
      - "4566"
    ports:
      - "4566:4566"
    depends_on: []
```

**Comment by skyzh** [comment]
Still testing...

File: src/risedevtool/src/compose.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+impl Compose for MetaNodeConfig {
+    fn compose(&self) -> Result<ComposeService> {
+        let mut command = Command::new("/risingwave/bin/meta-node");
+        MetaNodeService::apply_command_args(&mut command, self)?;
+        let command = get_cmd_args(&command)?;
+
+        Ok(ComposeService {
+            image: "ghcr.io/singularity-data/risingwave:latest".into(),
```
Comment (neverchanje) [inline src/risedevtool/src/compose.rs:95]: Better to use a constant.

File: src/risedevtool/src/compose.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                self.dashboard_port.to_string(),
+            ],
+            ..Default::default()
+        })
+    }
+}
+
+impl Compose for FrontendConfig {
+    fn compose(&self) -> Result<ComposeService> {
+        let mut command = Command::new("/risingwave/bin/frontend-v2");
```
Comment (neverchanje) [inline src/risedevtool/src/compose.rs:109]:
```suggestion
        let mut command = Command::new(FRONTEND_BIN_PATH);
```

**Review by neverchanje — APPROVED** [review]
lgtm

**Comment by skyzh** [comment]
Auto merge enabled 🤣 will fix in future PRs.

File: src/risedevtool/src/compose.rs (hunk 2 above)
Comment (skyzh) [reply src/risedevtool/src/compose.rs:109]: In https://github.com/singularity-data/risingwave/pull/2299/, I've specified the path of binary in Dockerfile, so we won't need to specify here in the future.

**Pull request merged**